src/Symfony/Bundle/DependencyInjection/Configuration.php

```diff
@@ -305,45 +306,185 @@ private function addSwaggerSection(ArrayNodeDefinition $rootNode): void
+                                break;
+                            }
+
+                            $config['tags']['invalidator']['banner'] = [
+                                'urls' => $config['invalidation']['varnish_urls'],
+                                'header' => [
+                                    'max_size' => $config['invalidation']['max_header_length'] ?? 7500,
+                                ],
+                                'client_options' => $config['invalidation']['request_options'] ?? [],
+                            ];
+
+                            break;
+                        default:
+                            $config['tags']['invalidator']['id'] = $config['invalidation']['purger'];
+                    }
+
+                    unset($config['invalidation']);
+
+                    return $config;
+                })
+            ->end()
+            ->children()
+                ->arrayNode('invalidation')
+                    ->setDeprecated('api-platform/core', '3.1', 'The child node "%node%" at path "%path%" is deprecated, use "tags" instead.')
+                    ->info('Enable the tags-based cache invalidation system.')
+                    ->canBeEnabled()
+                    ->children()
+                        ->arrayNode('varnish_urls')
+                            ->defaultValue([])
+                            ->prototype('scalar')->end()
+                            ->info('URLs of the Varnish servers to purge using cache tags when a resource is updated.')
+                        ->end()
+                        ->integerNode('max_header_length')
+                            ->defaultValue(7500)
+                            ->info('Max header length supported by the server')
+                        ->end()
+                        ->variableNode('request_options')
+                            ->defaultValue([])
+                            ->validate()
+                                ->ifTrue(static fn($v): bool => false === \is_array($v))
+                                ->thenInvalid('The request_options parameter must be an array.')
+                            ->end()
+                            ->info('To pass options to the client charged with the request.')
+                        ->end()
+                        ->scalarNode('purger')
+                            ->defaultValue('api_platform.http_cache.purger.varnish')
+                            ->info('Specify a varnish purger to use (available values: "api_platform.http_cache.purger.varnish.ban" or "api_platform.http_cache.purger.varnish.xkey").')
+                        ->end()
+                        ->arrayNode('xkey')
+                            ->addDefaultsIfNotSet()
+                            ->children()
+                                ->scalarNode('glue')
+                                ->defaultValue(' ')
+                                ->info('xkey glue between keys')
```

Comment:
I thought that we discussed about the configuration to have less components. Indeed, we want this to be configured using urls and/or scoped http clients (to support auth, default headers and complex approaches). The glue should be configured directly in the service as it is provider-specific. Note that it'd be a valid Symfony configuration to do:

```
services:
  App\MyPurger: 
     parent: 'api_platform.purger'
     arguments: 
         - !tagged api_platform.purger.http_client
         - ', ' // my own glue
         - 'Cache-Tags' // my own header 
```